toolkit/tools/scheduler/schedulerutils/printresults.go

```diff
@@ -33,6 +35,108 @@ func PrintBuildResult(res *BuildResult) {
 	}
 }
 
+// RecordBuildSummary stores the summary in to a csv.
+func RecordBuildSummary(pkgGraph *pkggraph.PkgGraph, graphMutex *sync.RWMutex, buildState *GraphBuildState, outputPath string) {
+
+	graphMutex.RLock()
+	defer graphMutex.RUnlock()
+
+	failedSRPMs := make(map[string]*pkggraph.PkgNode)
+	failures := buildState.BuildFailures()
+	for _, failure := range failures {
+		failedSRPMs[failure.Node.SrpmPath] = failure.Node
+	}
+
+	prebuiltSRPMs := make(map[string]*pkggraph.PkgNode)
+	builtSRPMs := make(map[string]*pkggraph.PkgNode)
+	unbuiltSRPMs := make(map[string]*pkggraph.PkgNode)
+	unresolvedDependencies := make(map[string]bool)
+
+	buildNodes := pkgGraph.AllBuildNodes()
+	for _, node := range buildNodes {
+		if buildState.IsNodeCached(node) {
+			prebuiltSRPMs[node.SrpmPath] = node
+			continue
+		} else if buildState.IsNodeAvailable(node) {
+			builtSRPMs[node.SrpmPath] = node
+			continue
+		}
+
+		_, found := failedSRPMs[node.SrpmPath]
+		if !found {
+			unbuiltSRPMs[node.SrpmPath] = node
+		}
+	}
+
+	for _, node := range pkgGraph.AllRunNodes() {
+		if node.State == pkggraph.StateUnresolved {
+			unresolvedDependencies[node.VersionedPkg.String()] = true
+		}
+	}
+
+	csvBlob := [][]string{{"Package", "State", "Blocker"}}
+
+	for srpm := range builtSRPMs {
+		csvBlob = append(csvBlob, []string{filepath.Base(builtSRPMs[srpm].SrpmPath), "Built"})
+	}
+
+	for srpm := range prebuiltSRPMs {
+		csvBlob = append(csvBlob, []string{filepath.Base(prebuiltSRPMs[srpm].SrpmPath), "PreBuilt"})
+	}
```

Comment:
We can access the values directly and simplify the code a bit:
```suggestion
	for _, srpm := range builtSRPMs {
		csvBlob = append(csvBlob, []string{filepath.Base(srpm.SrpmPath), "Built"})
	}

	for _, srpm := range prebuiltSRPMs {
		csvBlob = append(csvBlob, []string{filepath.Base(srpm.SrpmPath), "PreBuilt"})
	}
```